### DISCUSSION TITLE
Need to wrap app in Provider in mobx 6?

### DISCUSSION THREAD

**Discussion by idanlevi1** [mobxjs/mobx#2877] [Q&A]
I moved to mobx 6v & mobx-react 7v
and I want to ask if I need to wrap root node in Provider like before:
```
<Provider {...stores}>
      <Router>
        <Switch>
          <Route path='/' exact component={RootComponent} />
        </Switch>
      </Router>
</Provider>
```

for example

> "mobx": "^6.1.8",
> "mobx-persist-store": "^0.5.30"
> "mobx-react": "^7.1.0"

**Comment by mweststrate** [comment]
If you are still using inject, yes, otherwise no.

**Reply by idanlevi1** [reply]
> If you are still using inject, yes, otherwise no.
> […](#)
> On Wed, 7 Apr 2021, 18:05 Idan Levi, ***@***.***> wrote: I moved to mobx 6v & mobx-react 7v and I want to ask if I need to wrap root node in Provider like before: <Provider {...stores}> <Router> <Switch> <Route path='/' exact component={RootComponent} /> </Switch> </Router> </Provider> for example "mobx": "^6.1.8", "mobx-persist-store": "^0.5.30" "mobx-react": "^7.1.0" — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <#2877>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAN4NBGF2SGCOK5AXHUD3MDTHSGGJANCNFSM42RFWAPQ> .

How to use mobx without `inject`? I'm still using inject, maybe I can remove it...

**Reply by kubk** [reply]
@idanlevi1 It's documented here: https://mobx.js.org/react-integration.html#local-and-external-state

Click the third tab "using React context". I believe Mobx should have a dedicated page in docs describing how to use React context with Mobx because lots of people struggle with it. But my English level doesn't allow me to write documentation 😞
